util/sha256x/sha256.go

```diff
@@ -24,13 +25,16 @@ type Hash struct {
 	// However, it does mean that sha256.digest.x goes unused,
 	// which is a waste of 64B.
 
-	h  hash.Hash              // always *sha256.digest
+	// H is the underlying hash.Hash.
+	// The hash.Hash.BlockSize must be equal to sha256.BlockSize.
+	// It is exported only for testing purposes.
+	H  hash.Hash              // usually a *sha256.digest
```

Comment:
How about keep this unexported by add a package-level exported variable of unexported type, like `util/clientmetric` does next door to this directory:

```
util/clientmetric/clientmetric.go:var TestHooks testHooks
util/clientmetric/clientmetric.go:type testHooks struct{}
util/clientmetric/clientmetric.go:func (testHooks) ResetLastDelta() {
```

Then add a method like `(testHooks).SetHashUnderlyingHash(*Hash, hash.Hash)` and have the caller test code type assert that on the `testHooks` type.

Keeps the API clean looking.

Reply:
I think I'm going to export this for general use in the future.

I'm realizing that `sha256x` is probably name since this isn't actually about `sha256`, but rather a concrete implementation of `hash.Hash` with a particular block size.

According to https://en.wikipedia.org/wiki/Comparison_of_cryptographic_hash_functions#Parameters, common hash functions have the following block sizes (in bytes):
* 32B: GOST, PANAMA
* 64B: BLAKE2s, BLAKE3, MD4, MD5, RIPEMD, RIPEMD-128, RIPEMD-256, RIPEMD-160, RIPEMD-320, SHA-0, SHA-1, SHA-224, SHA-256, Tiger(2), Whirpool
* 128B: BLAKE2b, HAVAL, SHA-384, SHA-512

However, SHA3 is really odd in that it doesn't have a block size that's a power of 2. Yuck.

Reply:
Right now, the `Hash` type has a hardcoded block of 64B, which makes it suitable for most of the crypto hashes. If we increase it to 128B, it would be suitable for all crypto hashes except SHA3.

Reply:
But you still submitted this saying "It is exported only for testing purposes." when I suggested an alternative way to test.